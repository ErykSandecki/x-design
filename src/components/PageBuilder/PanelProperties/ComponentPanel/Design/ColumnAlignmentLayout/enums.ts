export enum PopoverItem {
  fixed = 0,
}
