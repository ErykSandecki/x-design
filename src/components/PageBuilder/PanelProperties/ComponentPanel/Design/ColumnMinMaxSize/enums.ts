export enum PopoverItem {
  currentValue = 0,
  auto = 1,
  unit = 2,
  removeScore = 3,
}
