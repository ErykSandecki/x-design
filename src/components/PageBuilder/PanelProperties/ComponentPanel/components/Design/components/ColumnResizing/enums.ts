export enum PopoverItem {
  auto = 1,
  fixed = 0,
  maxScore = 4,
  minScore = 3,
  unit = 2,
}
