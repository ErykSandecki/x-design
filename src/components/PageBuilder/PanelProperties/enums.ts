export enum Tab {
  design = 'design',
  prototype = 'prototype',
}
