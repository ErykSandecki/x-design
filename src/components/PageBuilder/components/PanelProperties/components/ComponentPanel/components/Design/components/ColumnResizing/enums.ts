export enum PopoverItem {
  auto = 1,
  fixed = 0,
  unit = 2,
}
