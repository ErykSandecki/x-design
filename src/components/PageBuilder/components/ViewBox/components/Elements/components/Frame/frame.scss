@import '../../../../../../styles/z-indexes';

.Frame {
  background-color: #ffffff;

  &__wrapper {
    pointer-events: none;
    position: absolute;
  }

  &__label {
    @include theme('neutral-2', color);

    font-size: 11px;
    pointer-events: all;
    position: absolute;
    transform: translate(1.5px, -20px);
    user-select: none;

    &--hover {
      @include theme('blue-1', color);
    }

    &--selected {
      @include theme('blue-1', color);
    }
  }
}
