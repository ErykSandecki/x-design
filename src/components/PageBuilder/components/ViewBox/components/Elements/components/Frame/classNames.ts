import { values } from 'lodash';

// types
import { ElementStickWallPosition } from 'components/PageBuilder/components/ViewBox/types/enums';

// utils
import { composeClassNames } from 'utils';

export const className = 'Frame';

export const classNames = composeClassNames(
  className,
  [className] as const,
  ['wrapper'] as const,
  ['label', 'hover', 'selected', ...values(ElementStickWallPosition)] as const,
);
