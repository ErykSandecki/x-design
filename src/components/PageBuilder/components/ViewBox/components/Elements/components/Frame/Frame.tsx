import { createPortal } from 'react-dom';
import { FC, memo, useMemo } from 'react';
import { useTranslation } from 'react-i18next';

// components
import Element from '../Element/Element';
import Elements from '../../Elements';
import { Box, Small } from 'shared';

// core
import { useRefs } from 'pages/PageBuilderPage/core/RefsProvider';

// hooks
import { useTheme } from 'hooks';

// others
import { className as classNameFrame, classNames } from './classNames';
import { translationNameSpace } from './contants';

// styles
import styles from './frame.scss';

// types
import { TElementProps } from '../../types';

// utils
import { getElementStickWallPosition } from 'components/PageBuilder/components/ViewBox/utils/getElementStickWallPosition';

export type TFrameProps = TElementProps;

const Frame: FC<TFrameProps> = ({ className, id, index, mouseMode, parentId, type }) => {
  const { overlayContainerRef } = useRefs();
  const { classNamesWithTheme, cx } = useTheme(classNames, styles);
  const { t } = useTranslation();

  return (
    <Element
      classes={{
        className: cx(className, classNamesWithTheme[classNameFrame]),
      }}
      id={id}
      index={index}
      parentId={parentId}
      mouseMode={mouseMode}
      type={type}
    >
      {(angle, coordinates, height, hover, selected, width) => {
        const stickWall = useMemo(() => getElementStickWallPosition(angle), [angle]);

        return (
          <>
            {overlayContainerRef.current &&
              parentId === '-1' &&
              createPortal(
                <Box
                  classes={{ className: cx(classNamesWithTheme.wrapper) }}
                  style={{
                    height: `${height}px`,
                    left: `${coordinates.x}px`,
                    top: `${coordinates.y}px`,
                    transform: `rotate(${angle}deg)`,
                    width: `${width}px`,
                  }}
                >
                  <Small
                    classes={{
                      className: cx(
                        classNamesWithTheme.label.name,
                        [classNamesWithTheme.label.modificators.hover, hover],
                        [classNamesWithTheme.label.modificators.selected, selected],
                        classNamesWithTheme.label.modificators[stickWall],
                      ),
                    }}
                  >
                    {t(`${translationNameSpace}.label.createFrame`)}
                  </Small>
                </Box>,
                overlayContainerRef.current,
              )}
            <Elements eventsDisabled={false} isSelected={selected} mouseMode={mouseMode} parentId={id} />
          </>
        );
      }}
    </Element>
  );
};

export default memo(Frame);
