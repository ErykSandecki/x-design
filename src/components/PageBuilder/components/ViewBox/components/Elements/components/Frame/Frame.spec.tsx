import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

// components
import Frame from './Frame';

// core
import { RefsProvider } from 'pages/PageBuilderPage/core/RefsProvider';

// mocks
import {
  elementAllDataMock,
  elementDynamicDataMock,
  elementStaticDataMock,
  pageBuilderStateMock,
} from 'test/mocks/reducer/pageBuilderMock';
import { wholeStateMock } from 'test/mocks/reducer/wholeStateMock';
import 'test/mocks/sagas/allSagas';

// others
import { REDUCER_KEY as PAGE_BUILDER } from 'store/pageBuilder/actionsType';

// store
import { configureStore } from 'store/store';

// types
import { MouseMode } from 'types/enums/mouseMode';

const element = document.createElement('div');
const overlayContainer = document.createElement('div');

const sharedRefs = {
  [elementAllDataMock.id]: element,
};

const stateMock = {
  ...wholeStateMock,
  [PAGE_BUILDER]: {
    ...pageBuilderStateMock[PAGE_BUILDER],
    pages: {
      ...pageBuilderStateMock[PAGE_BUILDER].pages,
      ['0']: {
        ...pageBuilderStateMock[PAGE_BUILDER].pages['0'],
        elements: {
          ...pageBuilderStateMock[PAGE_BUILDER].pages['0'].elements,
          dynamicData: {
            ...pageBuilderStateMock[PAGE_BUILDER].pages['0'].elements.dynamicData,
            [elementDynamicDataMock.id]: elementDynamicDataMock,
          },
          staticData: {
            ...pageBuilderStateMock[PAGE_BUILDER].pages['0'].elements.staticData,
            [elementStaticDataMock.id]: elementStaticDataMock,
          },
        },
      },
    },
  },
};

describe('Frame snapshots', () => {
  beforeAll(() => {
    element.style.height = '100px';
    element.style.width = '100px';
    document.body.appendChild(overlayContainer);
  });

  it('should render Frame', () => {
    // mock
    const store = configureStore(stateMock);

    // before
    const { asFragment } = render(
      <Provider store={store}>
        <RefsProvider itemsRefs={sharedRefs} overlayContainerRefHtml={overlayContainer}>
          <Frame
            className="className"
            id={elementDynamicDataMock.id}
            index={0}
            mouseMode={MouseMode.default}
            parentId={elementStaticDataMock.parentId}
            type={elementStaticDataMock.type}
          />
        </RefsProvider>
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});
