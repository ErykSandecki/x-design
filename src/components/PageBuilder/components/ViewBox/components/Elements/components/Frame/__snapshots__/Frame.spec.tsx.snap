// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Frame snapshots should render Frame 1`] = `
<DocumentFragment>
  <div
    class="css-wcnq0t className Frame Frame--light Element Element--light"
    data-test-box=""
    id="1"
    style="left: 0px; top: 0px; transform: rotate(0deg) translate(0, 0); transform-origin: center center; height: 100px; position: absolute; width: 100px;"
  >
    <div
      class="css-wcnq0t className Frame Frame--light Element__wrapper Element__wrapper--light"
      data-status="false"
      data-test-box=""
      style="background-color: rgb(255, 255, 255); height: 100%; transform: rotate(0deg); width: 100%;"
    >
      <div
        class="css-wcnq0t Frame__wrapper Frame__wrapper--light"
        data-test-box=""
        style="height: 100px; left: 0px; top: 0px; transform: rotate(0deg); width: 100px;"
      >
        <small
          class="css-wcnq0t Frame__label Frame__label--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
          data-test-text=""
          style="text-align: inherit;"
        >
          Frame
        </small>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
